shim/test/realm/test-repair.js

```diff
@@ -54,6 +54,30 @@ test('fix the bug in which accessor methods leak the global', t => {
   t.end();
 });
 
+function getGenerator() {
+  function* aStrictGenerator() {
+    yield;
+  }
+  return Object.getPrototypeOf(aStrictGenerator);
+}
+
+test('strict-function', t => {
+  const r = Realm.makeRootRealm();
+  const c = r.evaluate('Function.prototype.constructor');
+  t.notOk('arguments' in Object.getOwnPropertyDescriptors(c));
+  t.notOk('caller' in Object.getOwnPropertyDescriptors(c));
+  t.end();
+});
+
+test('generator-constructor-is-consistent', t => {
+  const r = Realm.makeRootRealm();
+  const c = r.evaluate('Function.prototype.constructor');
+  const gf = r.evaluate(`(${getGenerator})`)();
```

Comment:
I don't think this variable should be called "gf" nor the next one "gfc". This `gf` variable is bound to the `%Generator%` intrinsic. `gfc` is bound to the `%GeneratorFunction%` intrinsic, which is the same as `%Generator%.constructor`.

Reply:
how about `gp` and `gpc`? They're basically the prototype of a generator, and that prototype's constructor.

Reply:
y. `gp` and `gpc` are good